src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js

```diff
@@ -43,6 +43,17 @@ export class DateTimeSelectPageObject extends PageObject {
     return this;
   }
 
+  compareDatesClickNextMonth(firstDate, secondDate) {
+    if (firstDate.month() < secondDate.month()) {
+      cy.contains('button', 'Next')
+        .as('button')
+        .should('not.be.disabled')
+        .focus();
+      cy.get('@button').click();
+    }
+    return this;
+  }
+
```

Comment:
I added the changes to the 'clickNextMonth' because that method is called in multiple tests scenarios. The thought was that this check should always happen. I think this new method would have to be called everywhere the DateTimeSelectPageObject.clickNext is used.

Reply:
@vbahinwillit , There are 2 scenario that call for `DateTimeSelectPageObject.clickNextMonth` The first is, "should fetch slots when moving between months" When adding the end of month test for an additional click to `DateTimeSelectPageObject.clickNextMonth`,  it breaks the assertion count.  Instead of expected count of 3, it will be 4. Advancing to the next month is not needed in this situation.

The second scenario is “should allow a user to choose available slot and fetch new slots after changing clinics”. `DateTimeSelectPageObject.compareDatesClickNextMonth` was created to test for 2 days prior to end of month. Because same day appointment is not allowed, the next available date (firstDate) is tomorrow and the second available date is day after tomorrow. If secondDate lands on the next month, it clicks next month to find the secondDate.  At the present time, this is the only scenario that needs the 2 days prior end of month check.

 I think these 2 scenarios require different treatment for click next month.